src/main/java/com/mongodb/jdbc/MongoConnection.java

```diff
@@ -524,12 +557,15 @@ public SQLXML createSQLXML() throws SQLException {
     class ConnValidation implements Callable<Void> {
         @Override
         public Void call() throws SQLException {
-            Statement statement = createStatement();
-            boolean resultExists = statement.execute("SELECT 1");
-            if (!resultExists) {
-                // no resultSet returned
-                throw new SQLException("Connection error");
+            MongoClusterType actualClusterType = determineClusterType();
+
+            if (actualClusterType == MongoClusterType.Community) {
+                // Community edition is disallowed
+                throw new SQLException("Community edition detected. The JDBC driver is intended for use with MongoDB Enterprise edition or Atlas Data Federation.");
             }
+
+            // Set the cluster type.
+            clusterType = actualClusterType;
```

Comment:
@EthanHardyMongo can you clarify if anything else needs to be done as part of this ticket?

Specifically, am I supposed to check that the library is loaded if we discover an enterprise cluster? I'm not sure what the table in the design doc means when it says "Library Available: N". What does the "N" stand for here? ([link](https://docs.google.com/document/d/1rpKyQ-DsPbWsNOMJsp2cp1dKm99lvzp7_uj2HmOHfFI/edit#bookmark=id.h0s8zd1atopw))

Is there anything else to do here besides check that the library is loaded?

Reply:
I updated it to throw an exception if the library is not loaded in enterprise mode. Let me know if there is anything else that's supposed to be added!

Reply:
This looks good to me in order to notify users that there was an issue with loading the library.

Reply:
@mattChiaravalloti Sorry, in retrospect, the table was not very clear. Natacha updated it to make more sense. But to answer your questions, the `N` was supposed to mean `No`. There's now a `Y` section to make this more clear. And yes, if you are in enterprise mode and the library is not available (this information is discovered before connection time and should be available) then you would throw an error.